update roadmap 2023 + how to contribute

actuallyyun opened the pull request:
### Roadmap
- Anything you'd like to add?
- Add items to the `By the end of 2023` list
- Should we have a page about the climbing log project so people can find relevant info(or, current project page) and link the page to each roadmap item

actuallyyun committed:
update roadmap 2023 draft1

actuallyyun committed:
update how to contribute draft 1

actuallyyun committed:
add general onboarding

actuallyyun committed:
add dev onboarding tasks

vnugent commented:
can you have a look at the build log?

``` 

Error: Markdown link URL is mandatory in "docs/how-to-contribute/dev-onboarding.md" file (title: general onboarding, line: 2).
--
07:57:58.164 | Error: Markdown link URL is mandatory in "docs/how-to-contribute/general-onboarding.md" file (title: our blog, line: 10).
07:57:58.164 | Error: Markdown link URL is mandatory in "docs/how-to-contribute/overview.md" file (title: Code of Conduct, line: 114).

```

actuallyyun commented:
> can you have a look at the build log?
> 
> ```
> 
> Error: Markdown link URL is mandatory in "docs/how-to-contribute/dev-onboarding.md" file (title: general onboarding, line: 2).
> --
> 07:57:58.164 | Error: Markdown link URL is mandatory in "docs/how-to-contribute/general-onboarding.md" file (title: our blog, line: 10).
> 07:57:58.164 | Error: Markdown link URL is mandatory in "docs/how-to-contribute/overview.md" file (title: Code of Conduct, line: 114).
> ```

Yeah because we don't have these pages yet so I left the url empty. I pushed the changes as a draft to get feedback from you guys.

CocoisBuggy commented on docs/how-to-contribute/general-onboarding.md:
```diff
@@ -0,0 +1,65 @@
+# Step 2 Deep Dive About Open Beta
+
+- [ ] Read [our roadmap](https://docs.openbeta.io/introduction/roadmap)
+- [ ] Read [our blog]()
```
https://openbeta.substack.com/

I saw you said you were aware that the links aren't filled, but just thought I'd leave this here in case you were looking for it

CocoisBuggy commented on docs/how-to-contribute/overview.md:
This came out really well. I like the concept for the titling 😁

vnugent commented:
@actuallyyun Go for it.  We can revise again.

One thing about naming open-tacos vs climb catalog:
https://github.com/OpenBeta/docs.openbeta.io/issues/20

musoke commented on docs/how-to-contribute/general-onboarding.md:
```diff
@@ -0,0 +1,65 @@
+### GitHub Issues:
+
+- Use GitHub issues to report bugs, suggest features, or discuss project-related topics.
+- If you'd like to work on the issue, assign yourself.
```
Are permissions needed to assign issues?

vnugent replied:
Users can't assign the issue to themselves unless they're part of the org or are given permission.  You can leave a comment, I want to work on this issue.

musoke commented on docs/how-to-contribute/general-onboarding.md:
```diff
@@ -0,0 +1,65 @@
+
+### Pull Requests (PRs):
+
+- Once you know how to start working on the issue, you could fork the project repository and create a branch for your changes. 
```
could -> should?

musoke commented on docs/how-to-contribute/general-onboarding.md:
```diff
@@ -0,0 +1,65 @@
+### Pull Requests (PRs):
+
+- Once you know how to start working on the issue, you could fork the project repository and create a branch for your changes. 
+- Submit PRs to propose changes, bug fixes, or new features.
```
"Each PR should fix a single issue or small set of related issues."

musoke commented on docs/how-to-contribute/general-onboarding.md:
```diff
@@ -0,0 +1,65 @@
+
+- Once you know how to start working on the issue, you could fork the project repository and create a branch for your changes. 
+- Submit PRs to propose changes, bug fixes, or new features.
+- Request reviews from our core dev team members, engage in PR discussions, respond to feedback, and iterate on the changes based on reviewer comments.
```
I would add guidance on who to ask for a review.  It can be intimidating and confusing for a first time contributor to decide who to ask.  Worst case scenario, github's suggested reviewers are disengaged from the and a new contributor's PR languishes forever.  For example,

> If you're not sure who should review your changes, ask @.... or @... and they will direct PR to an appropriate reviewer.

Maybe @actuallyyun would be a good person for now, as the new dev onboarder.

actuallyyun replied:
Great idea!

musoke commented on docs/how-to-contribute/overview.md:
```diff
@@ -6,31 +6,91 @@ title: "Welcome"
 
-### Data Science & Data Engineering
+#### Code 
+⌛: 10 hrs/week
```
Is this meant to be less than 10 hrs?

actuallyyun replied:
Good catch!

musoke reviewed (approved):
Looks good!   I left a few comments, but none that should block this if others think they aren't a priority.

actuallyyun committed:
update overview and general onboarding based on PR reviews

actuallyyun committed:
update roadmap

actuallyyun committed:
reassign sidebar position

actuallyyun commented:
@vnugent revised based on the reviews. What's the merge workflow for org members?

actuallyyun committed:
Delete dev-onboarding.md

remove dev-onboarding from current PR to keep PR in scope. Will open a new one for dev onboarding.

actuallyyun committed:
Delete all-contributors.md

Remove from current PR. Will add in the next PR.

actuallyyun commented on docs/how-to-contribute/general-onboarding.md:
```diff
@@ -0,0 +1,72 @@
+## Synchronized Meetings
+We meet bi-weekly to synchronize efforts and provide updates on the project's progress. Check out the current project we are working on and join the meeting if you are interested.
+
+You could find meeting infomation in our discord channel.
```
typo infomation => information
discord => Discord

actuallyyun commented on docs/how-to-contribute/general-onboarding.md:
```diff
@@ -0,0 +1,72 @@
+
+- Everyone can open an issue. To work on an issue, you should leave a comment stating that you want to work on this issue, and our maintainers will assign you if appropriate.
+- If you'd like to work on an existing issue, please comment and make sure that no one is working on it to avoid duplicated effort.
+- We recommend you discuss the solutions you have in mind before you start working on it. If you don't know where to start, you should ask for help. Please refer to the organization workflow to start a conversation. 
```
recommend you discuss => recommend you to discuss

actuallyyun commented on docs/how-to-contribute/overview.md:
```diff
@@ -6,31 +6,90 @@ title: "Welcome"
+
+🔨: none
+
+If you have limited time but still want to contribute, you can help by simply start using open beta 
```
open beta => Open Beta, our climbing catalog.

actuallyyun commented on docs/how-to-contribute/overview.md:
```diff
@@ -6,31 +6,90 @@ title: "Welcome"
+If you have limited time but still want to contribute, you can help by simply start using open beta 
+
+1. Register an account, browse climbing data, add climbing data if you can't find what you are looking for, and share your climbing photos with the community. 
+2. You could also support us by following our social media accounts, and tagging us when you post something about open beta.
```
Open Beta

actuallyyun commented on docs/how-to-contribute/overview.md:
```diff
@@ -6,31 +6,90 @@ title: "Welcome"
-We want contributing to the project be fun and educational. Here's a non-exhaustive list of areas we're needing help with:
+🔨: UI/UX Marketing Communication
 
-### Data maintenance and growth using [OpenTacos](/how-to-contribute/using-opentacos/introduction)
```
OpenTacos => Open Beta

actuallyyun commented on docs/how-to-contribute/overview.md:
```diff
@@ -46,28 +105,19 @@ We ask that you fork the repo you will be working on, and submit pull requests f
 :::
 
-**Review the [onboarding checklist](./onboarding-checklist).**
+➡️ You can join our core dev team by simply completing our dev onboarding program and work on some tasks first. If by then you are still interested joining, please express your interest in our Discord channel.
```
interested in joining

actuallyyun commented on docs/introduction/roadmap.md:
```diff
@@ -3,28 +3,25 @@ sidebar_position: 3
-### Medium Term (due August 2022)
+- Climbing Log Project
+- Upgrade database to support multipitch routes
 
```
Improve contributors onboarding experience

actuallyyun committed:
fix typo and wording
